assets/src/components/Sidebar/NewThemeSettings.js

```diff
@@ -0,0 +1,87 @@
+import { SelectControl } from '@wordpress/components';
+import { useState, useEffect } from 'react';
+import { p4ServerThemes } from '../../theme/p4ServerThemes';
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+
+const keysAsLabel = obj => Object.keys(obj).map(k => ({ label: k, value: k }));
+
+// Just a lightweight way to have a separate UI element in the sidebar so that it's clear which themes are legacy and
+// which are new. This is likely just for internal usage to facilitate reviewing the refactored version, eventually
+// editors will work with a single select.
+const useServerThemes = () => {
+  const [serverThemes, setServerThemes] = useState({});
+
+  useEffect(() => {
+    const fetchThemes = async () => {
```

Comment:
You can implement an IIFE function, i think it's a better solutions instead of creating a new function variable
```
(async () => {
    const themes = await p4ServerThemes.fetchThemes();
     setServerThemes(themes);
})();

```
Not necessary to return a value. `useEffect` returns `void`. You will be able to take `serverThemes` value from its state

Reply:
Nice didn't think of IIFE :ok_hand: 

> Not necessary to return a value

Do you mean this line? https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/589/files#diff-a281525a512c25313074cd59d3954135907ee63aa7045f1ce29c686a9a291b2cR23

Reply:
Perfect!

Yes, that line!

Reply:
That is the return statement of the custom hook `useServerThemes`